Automattic/newspack-plugin
# feat: lock access to My Account UI until account is verified

### All Submissions:






### Changes proposed in this Pull Request:



Locks access to My Account pages until the account is verified. This only applies to reader accounts that haven't been verified via SSO login, password reset, or magic link authentication.

### How to test the changes in this Pull Request:

1. Check out this branch.
2. In a new session, register a new account via direct email input.
3. Visit My Account and confirm that you only see menu items for "Account Details" and "Log out", and that Account Details shows the following messaging and CTAs:

<img width="1212" alt="Screen Shot 2022-08-09 at 3 20 41 PM" src="https://user-images.githubusercontent.com/2230142/183764172-b42936eb-38bb-45f2-bead-b9478e9d38ab.png">

4. Also confirm that if you try to visit another My Account page via direct URL (e.g. `https://sitename/my-account/newsletters/`), that you're redirected to the "Account Details" verification page.
5. Click "Send me a link" and confirm that you get a verification email, and that visiting the URL from the verification email unlocks the normal My Account UI with a message thanking you for verifying.
6. Repeat steps 2-3, but this time click "Set a new password" and follow the flow for resetting your password. Then visit My Account and confirm that the normal My Account UI is unlocked.
7. As a failsafe, disable or delete the `NEWSPACK_EXPERIMENTAL_READER_ACTIVATION` flag in `wp-config.php` and make a donation via the Donate block.
8. Visit My Account and confirm that My Account pages are not locked if Reader Activation features aren't enabled.

### Other information:
